docs/src/motion/pid-theory.adoc

```diff
@@ -188,4 +188,98 @@ After tuning the axis check the following error with Halscope to make
 sure it is within your machine requirements. More information on
 Halscope is in the HAL User manual.
 
+=== Automatic PID tuning
+
+Since LinuxCNC version 2.9, the pid component support automatic tuning
+using the Relay method footnote:[The relay method is described in the
+1984 paper Automation paper Automatic Tuning of Simple Regulators with
+Specifications on Phase and Amplitude Margins by Karl Johan Åström and
+Tore Hägglund,
+link:https://doi.org/10.1016/0005-1098(84)90014-1[DOI10.1016/0005-1098(84)90014-1]].
+This is a replacement for the now removed and obsolete at_pid
+component.
+
+The pid component uses several constants to calculate the output value
+based on current and wanted state, the most important among them being
+Pgain', 'Igain', 'Dgain', 'bias', 'FF0', 'FF1', 'FF2' and 'FF3'. All
+of these need to have a sensible value for the controller to behave
+properly.
+
+The current implementation of automatic tuning implement two different
+algorithms, seleged using the tune-type pin.  When tune-type is zero,
+it affect 'Pgain', 'Igain' and 'Dgain' while setting 'FF0', 'FF1' and
+'FF2' to zero.  If tune-type is 1, it affect 'Pgain', 'Igain' and
+'FF1' while setting 'Dgain', 'FF0' and 'FF2' to zero.  Note type 1
+require scaling be set so output is in user units per second.
+
+When autotuning a motor with tupe-type 0, the algorithm will produce a
+square wave pattern centered around the 'bias' value on the output pin
+of the PID controller, moving from the positive extreme to the
+negative extreme of the output range.  This can be seen using the HAL
```

Comment:
Might be good to mention the `ferror` limits here also, even though the script does extend them automatically. The `ferror` can also help to automatically stop if the PID tuning goes way out of range, to prevent hitting any endstops.